### PR TITLE
Update window.ethereum type to unknown

### PR DESCRIPTION
**Pull request by erin-at-work**
### *Summary*

- Prevent conflicting wallet providers for `window.ethereum`

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -12,7 +12,11 @@ declare global {
   interface Window {
     CoinbaseWalletSDK: typeof CoinbaseWalletSDK;
     CoinbaseWalletProvider: typeof CoinbaseWalletProvider;
-    ethereum?: CoinbaseWalletProvider;
+    /**
+     * Workaround possible compatibility issues in Angular
+     * Set as CoinbaseWalletProvider
+     */
+    ethereum?: any;
```
Comment (erin-at-work) [inline src/index.ts:19]: This should probably be `unknown`?

File: src/index.ts [hunk 2]
```diff
@@ -12,11 +12,7 @@ declare global {
   interface Window {
     CoinbaseWalletSDK: typeof CoinbaseWalletSDK;
     CoinbaseWalletProvider: typeof CoinbaseWalletProvider;
-    /**
-     * Workaround possible compatibility issues in Angular
-     * Set as CoinbaseWalletProvider
-     */
-    ethereum?: any;
+    ethereum?: unknown;
```
Comment (ljharb) [inline src/index.ts:15]: i wonder if we could instead declare `unknown | PreciseType` so that type hinting is more useful.

Comment (erin-at-work) [reply src/index.ts:15]: I had first attempted that, but it doesn't seem to get picked up. 
> // In a union an unknown absorbs everything

from https://github.com/microsoft/TypeScript/pull/24439

image

and `coinbaseWalletExtension`:
image

I'll add back a comment

Comment (ljharb) [reply src/index.ts:15]: ah ok, oof then.

Comment (erin-at-work) [reply src/index.ts:18]: @ljharb Thoughts on addressing this issue? https://github.com/coinbase/coinbase-wallet-sdk/issues/343

> All declarations of 'ethereum' must have identical modifiers

In order to fix the above error, the type would need to be updated to:

```ts
interface Window {
  ethereum: any | undefined;
}
```

But I hesitate to make changes based on a conflicting library's type defs

**Review by bangtoven — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for bangtoven @ 2022-03-25 19:27:40 UTC
User failed mfa authentication, see [go/mfa-help](http://go/mfa-help)

**Review by bangtoven — APPROVED** [review]

**Pull request merged**